doc/scapy/usage.rst

```diff
@@ -1361,21 +1361,21 @@ DNS Requests
 This will perform a DNS request looking for IPv4 addresses
 
     >>> ans = sr1(IP(dst="8.8.8.8")/UDP(sport=RandShort(), dport=53)/DNS(rd=1,qd=DNSQR(qname="secdev.org",qtype="A")))
-    >>> ans.an.rdata
+    >>> ans.an[0].rdata
     '217.25.178.5'
 
 **SOA request:**
 
     >>> ans = sr1(IP(dst="8.8.8.8")/UDP(sport=RandShort(), dport=53)/DNS(rd=1,qd=DNSQR(qname="secdev.org",qtype="SOA")))
-    >>> ans.ns.mname
+    >>> ans.an[0].mname
     b'dns.ovh.net.'
-    >>> ans.ns.rname
+    >>> ans.an[0].rname
     b'tech.ovh.net.'
 
 **MX request:**
 
     >>> ans = sr1(IP(dst="8.8.8.8")/UDP(sport=RandShort(), dport=53)/DNS(rd=1,qd=DNSQR(qname="google.com",qtype="MX")))
-    >>> results = [x.exchange for x in ans.an.iterpayloads()]
+    >>> results = [x.exchange for x in ans.an]
```

Comment:
That's annoying. Would you mind sharing the pcap?

Reply:
I thought I attached it. Sorry!
[oom-fd7de6e369ecf4bfaaea53ca533e59db679bb961.zip](https://github.com/secdev/scapy/files/11511007/oom-fd7de6e369ecf4bfaaea53ca533e59db679bb961.zip)

FWIW I dropped the timeout and managed to read that pcap but when I tried to call `repr` with that packet `python` got killed by the OOM killer (in a VM with 4GB RAM).

Reply:
Having taken a closer look it seems the issue boils down to a lot of small QRs utilizing the compression and pointing to a large bogus name. As far as I understand with the master branch it doesn't pass the sanity checks and gets wrapped into `Raw` pretty fast. I think those sanity checks could prevent it here too but on top of them scapy could also cap the number of elements in the packet list fields. That would probably somewhat resemble `tshark` which just stops printing those bogus QRs at some point.

Reply:
> those sanity checks could prevent it

Having said that those sanity checks prevent my bespoke `DNS.tcp_reassemble` from assembling OPENPGPKEY RRs usually sent using TCP because of their size. With this PR applied it seems to work because regardless of whether RRs are split or not they don't get wrapped into `Raw` and `DNS.tcp_reassemble` gets called. I have to admit now I'm not sure how far the sanity checks should go.

Reply:
Looks like it no longer OOMs. Thanks! (I'll try to fully unleash my DNS corpora once this PR is merged). 

I tried to collect my thoughts on `tcp_reassemble` in https://github.com/secdev/scapy/issues/4023 but that's a minor feature request and can safely be ignored. I'll try to open a PR if I figure out how to assemble those packets :-)